src/logic/currencyValues/store/actions/fetchCurrencyValues.js

```diff
@@ -1,36 +1,41 @@
 // @flow
+import { List } from 'immutable'
 import { batch } from 'react-redux'
 import type { Dispatch as ReduxDispatch } from 'redux'
 
 import fetchCurrencySelectedValue from '~/logic/currencyValues/store/actions/fetchCurrencySelectedValue'
-import { CURRENCY_SELECTED_KEY } from '~/logic/currencyValues/store/actions/saveCurrencySelected'
+import { setCurrencyBalances } from '~/logic/currencyValues/store/actions/setCurrencyBalances'
 import { setCurrencyRate } from '~/logic/currencyValues/store/actions/setCurrencyRate'
 import { setCurrencySelected } from '~/logic/currencyValues/store/actions/setCurrencySelected'
 import { AVAILABLE_CURRENCIES } from '~/logic/currencyValues/store/model/currencyValues'
+import { loadCurrencyValues } from '~/logic/currencyValues/store/utils/currencyValuesStorage'
 import type { GlobalState } from '~/store'
-import { loadFromStorage } from '~/utils/storage'
 
-export const fetchCurrencyValues = () => async (dispatch: ReduxDispatch<GlobalState>) => {
+export const fetchCurrencyValues = (safeAddress: string) => async (dispatch: ReduxDispatch<GlobalState>) => {
   try {
-    const currencyStored = await loadFromStorage(CURRENCY_SELECTED_KEY)
-
+    const currenciesStored = await loadCurrencyValues()
+    const currencyStored = currenciesStored[safeAddress]
```

Comment:
Same rule applies to these variables

Reply:
Also, why do you need to load the currency values every time the action is called from the localstorage?

Reply:
Because we need to "fetch" the currencies values for that address and those values are on the localStorage

Reply:
why do we need to save them?

Reply:
To save the settings of the users, like the currency that he want's for his safe, maybe we could just save this instead of the balances

Reply:
yeah we should only store the information we fetch again (e.g safe name set by user). If something can be fetched again we shouldn't store this

Reply:
Do you mind implementing this?

Reply:
will do it

Reply:
it's done